server/src/main/java/org/elasticsearch/index/shard/IndexShard.java

```diff
@@ -533,6 +538,7 @@ public void updateShardState(
                     : "a primary relocation is completed by the master, but primary mode is not active " + currentRouting;
 
                 changeState(IndexShardState.STARTED, "global state is [" + newRouting.state() + "]");
+                startedRelativeTimeInNanos = getRelativeTimeInNanos();
```

Comment:
Should we then not also capture the indexing time spent when started? This would enable us to discount indexing operations that happened while we were doing operations based recovery?

Reply:
My impression was that the recovery indexing time should be negligible, but I'll add that too 👍

Reply:
I think you are right that it is mostly negligible, but I worry about edge cases where the replay from primary takes an extraordinary amount of time, has seen hours in the past (which could be due to other problems, but still).

Reply:
I've just noticed that we don't take into account the recovery time in the stats https://github.com/elastic/elasticsearch/blob/be7c7415627377a1b795400fb8dfcc6cbdf0e322/server/src/main/java/org/elasticsearch/index/shard/InternalIndexingStats.java#L47-L51

Reply:
Absolutely and I think that is correct compared to our use too?

However, notice that actual indexing that occur during recovery will be included.